rocketmq-client4cpp/example/demo/AsyncProducer.cpp

```diff
@@ -0,0 +1,253 @@
+    {
+    }
+
+    int count()
+    {
+
+        long long now = MyUtil::getNowMs();
+        long long old = g_lastUpdateTime;
+        long long total = g_cnt_succ + g_cnt_fail;
+        if ((now - old) >= 1000)
+        {
+            if (__sync_bool_compare_and_swap(&g_lastUpdateTime, old, now))
+            {
+                long long time = now - old;
+                int tps = (int)((total - g_cnt_last) * 1.0 / time * 1000.0);
+                g_cnt_last = total;
+
+                MYDEBUG("[producer]succ: %lld, fail: %lld, TPS: %d\n",
+                    g_cnt_succ, g_cnt_fail, tps);
+            }
+        }
+    }
+
+    void onSuccess(SendResult& sendResult)
+    {
+        int cnt = __sync_fetch_and_add(&g_cnt_total, 1);
+        __sync_fetch_and_add(&g_cnt_succ, 1);
+        MYLOG("[%d]|succ|%s\n",  cnt, sendResult.toString().c_str());
+    }
+
+    void onException(MQException& e)
+    {
+        int cnt = __sync_fetch_and_add(&g_cnt_total, 1);
+        __sync_fetch_and_add(&g_cnt_fail, 1);
+
+        MYLOG("[%d]|fail|%s\n",  cnt, e.what());
+    }
+};
+
+int main(int argc, char *argv[]) {
+    if (argc < 2)
+    {
+        Usage(argv[0]);
+        return 0;
+    }
+
+    std::string namesrv = argv[1];
+    std::string group = "pg_test_group";
+    std::string topic = "topic_test";
+    int size = 32;
+    int count = 1000;
+
+    for (int i=2; i< argc; i++)
+    {
+        if (strcmp(argv[i],"-g")==0)
+        {
+            if (i+1 < argc)
+            {
+                group = argv[i+1];
+                i++;
+            }
+            else
+            {
+                Usage(argv[0]);
+                return 0;
+            }
+        }
+        else if (strcmp(argv[i],"-t")==0)
+        {
+            if (i+1 < argc)
+            {
+                topic = argv[i+1];
+                i++;
+            }
+            else
+            {
+                Usage(argv[0]);
+                return 0;
+            }
+        }
+        else if (strcmp(argv[i],"-n")==0)
+        {
+            if (i+1 < argc)
+            {
+                count = atoi(argv[i+1]);
+                i++;
+            }
+            else
+            {
+                Usage(argv[0]);
+                return 0;
+            }
+        }
+        else if (strcmp(argv[i],"-s")==0)
+        {
+            if (i+1 < argc)
+            {
+                size = atoi(argv[i+1]);
+                i++;
+            }
+            else
+            {
+                Usage(argv[0]);
+                return 0;
+            }
+        }
+        else if (strcmp(argv[i],"-w")==0)
+        {
+            if (i+1 < argc)
+            {
+                MyUtil::initLog(argv[i+1]);
+                i++;
+            }
+            else
+            {
+                Usage(argv[0]);
+                return 0;
+            }
+        }
+        else
+        {
+            Usage(argv[0]);
+            return 0;
+        }
+    }
+
```

Comment:
Best to translate these Chinese documentation to English so that we may have more developers from various backgrounds join easier.